app/models/order.rb

```diff
@@ -429,13 +431,35 @@ def self.orders_count_per_state(orders)
     orders.each { |key, value| orders[key] = value.count }
   end
 
-  def self.generate_gc_code
-    record = where(detail_type: "GoodCity").order("id desc").first
-    "GC-" + gc_code(record).to_s.rjust(5, "0")
+  def self.order_code_prefix(detail_type)
+    return "" if detail_type.blank?
+    return 'GC-' if detail_type.match(/goodcity/i)
+    return "S" if detail_type.match(/shipment/i)
+    return "C" if detail_type.match(/carryout/i)
+  end
+
+  def self.generate_code(detail_type)
+    prefix = order_code_prefix(detail_type)
+    splitter = prefix.length.to_i+1
+    reg = %r/^\d+$/
+
+    code = next_code(detail_type, splitter, reg) + 1
+    code = code.to_s.rjust(5, '0')
+    "#{prefix}#{code}"
+  end
+
+  def self.next_code(type, splitter, reg)
```

Comment:
```suggestion
  def self.next_code(detail_type, splitter, reg)
```

Avoid usage of the keyword `type`